From 86f1eb7210bd224f76875904994fbd9b838e9b64 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Sun, 30 Dec 2018 11:57:12 +0800
Subject: [PATCH] drm/rockchip: vop: update default pdaf work mode

rk1808 cascade must set default pdaf work mode, after all
the chip csi rx and tx enale, the userspace will change to
normal mode.

Change-Id: I0ce18dc944bd55dbdb0812737affb11104e2f7a8
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
index 79998f912021..4916aaad93af 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
@@ -3561,8 +3561,8 @@ static void vop_crtc_atomic_flush(struct drm_crtc *crtc,
 	spin_lock_irq(&crtc->dev->event_lock);
 	if (crtc->state->event) {
 		WARN_ON(drm_crtc_vblank_get(crtc) != 0);
-		WARN_ON(vop->event);
-
+		if (s->pdaf_work_mode != VOP_HOLD_MODE)
+			WARN_ON(vop->event);
 		vop->event = crtc->state->event;
 		crtc->state->event = NULL;
 	}
@@ -3621,7 +3621,7 @@ static void vop_crtc_reset(struct drm_crtc *crtc)
 	s->right_margin = 100;
 	s->top_margin = 100;
 	s->bottom_margin = 100;
-	s->pdaf_work_mode = VOP_NORMAL_MODE;
+	s->pdaf_work_mode = VOP_HOLD_MODE;
 	s->pdaf_type = VOP_PDAF_TYPE_VBLANK;
 }
 
-- 
2.35.3

